### PR TITLE
kubectl: Remove swagger 1.2 entirely.

### PR DESCRIPTION
**Pull request by apelisse**
**What this PR does / why we need it**:
Remove dead code since nothing is using swagger 1.2 anymore. This doesn't change any feature, it's just removing unused code.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: Follow up on #44589

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/discovery/discovery_client.go [hunk 1]
```diff
@@ -336,41 +327,6 @@ func (d *DiscoveryClient) ServerVersion() (*version.Info, error) {
 	return &info, nil
 }
 
-// SwaggerSchema retrieves and parses the swagger API schema the server supports.
-// TODO: Replace usages with Open API.  Tracked in https://github.com/kubernetes/kubernetes/issues/44589
-func (d *DiscoveryClient) SwaggerSchema(version schema.GroupVersion) (*swagger.ApiDeclaration, error) {
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/discovery/discovery_client.go]: We don't know if there is any client-go user depends on swagger 1.2. Could you send an email to kubernetes-dev to ask if anyone objects remove swagger 1.2 from client-go? We can proceed if no one objects.

Comment (apelisse) [reply staging/src/k8s.io/client-go/discovery/discovery_client.go]: Thanks for looking at it! Yeah, doing it right now.

Comment (apelisse) [reply staging/src/k8s.io/client-go/discovery/discovery_client.go]: No news? Shall we proceed?

Comment (caesarxuchao) [reply staging/src/k8s.io/client-go/discovery/discovery_client.go]: Let's proceed.

File: pkg/kubectl/validation/schema.go [hunk 2]
```diff
@@ -20,35 +20,11 @@ import (
...
 	ejson "github.com/exponent-io/jsonpath"
-	"github.com/golang/glog"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
-	"k8s.io/apimachinery/pkg/util/yaml"
-	apiutil "k8s.io/kubernetes/pkg/api/util"
 )
 
-// InvalidTypeError records information about an invalid type.
-type InvalidTypeError struct {
```
Comment (caesarxuchao) [inline pkg/kubectl/validation/schema.go]: A little bit surprised that we have an error type that's only used by the swagger.

**Comment by caesarxuchao** [comment]
lgtm. @apelisse could you get someone from the kubectl side to approve the change as well?

File: pkg/kubectl/validation/schema.go (hunk 2 above)
Comment (apelisse) [reply pkg/kubectl/validation/schema.go]: This are just convenience type used by Swagger 1.2. OpenAPi validation has redefined these types ...

**Pull request merged**